src/nexgen/command_line/ED_mrc_to_nexus.py

```diff
@@ -0,0 +1,290 @@
+from nexgen.nxs_utils.ScanUtils import calculate_scan_points
+from pathlib import Path
+from argparse import ArgumentParser
+import hdf5plugin
+
+
+def main():
+
+    parser = ArgumentParser(add_help=False)
+    parser.add_argument(
+        "--det_distance",
+        type=float,
+        default=None,
+        help="The sample-detector distance.",
+    )
+    parser.add_argument(
+        "-wl",
+        "--wavelength",
+        type=float,
+        default=None,
+        help="Incident beam wavelength, in A.",
+    )
+    parser.add_argument(
+        "--axis-start",
+        type=float,
+        default=None,
+        help="Rotation axis start position.",
+    )
+    parser.add_argument(
+        "--axis-inc",
+        type=float,
+        default=None,
+        help="Rotation axis increment.",
+    )
+    parser.add_argument(
+        "-e",
+        "--exp-time",
+        type=float,
+        help="Exposure time, in s.",
+    )
+    parser.add_argument(
+        "-bc",
+        "--beam-center",
+        type=float,
+        nargs=2,
+        help="Beam center (x,y) positions.",
+    )
+    parser.add_argument('input_files', nargs='+', help='List of input files')
+
+    args = parser.parse_args()
+
+    for file in args.input_files:
+        if not file.endswith('.mrc'):
+            raise ValueError('Not an mrc file!')
+
+    mrc_files = args.input_files
+    tot_imgs, out_file, angles = collect_data(mrc_files)
+    mdict = get_metadata(mrc_files[0])
+
+    attenuator = Attenuator(transmission=None)
+
+    if args.det_distance is not None:
+        det_distance = args.det_distance
+    else:
+        det_distance = mdict['cameraLength']
+    if args.wavelength is not None:
+        beam = Beam(args.wavelength)
+    else:
+        beam = Beam(mdict['wavelength'])
+    if args.axis_start is not None:
+        start_angle = args.axis_start
+    else:
+        start_angle = angles[0]
+    if args.axis_inc is not None:
+        increment = args.axis_inc
+    else:
+        increment = mdict['tiltPerImage']
+    if args.exp_time is not None:
+        exposure_time = args.exp_time
+    else:
+        exposure_time = mdict['integrationTime']
+    if args.beam_center is not None:
+        if len(args.beam_center) == 2:
+            x, y = args.beam_center
+            beam_center = (x, y)
+        else:
+            msg = 'Beam center requires two arguments'
+            raise ValueError(msg)
+    else:
+        beam_center = (mdict["beamCentreXpx"],
+                       mdict["beamCentreYpx"])
+
+    gonio_axes = EDCeta.gonio
+    gonio_axes[0].start_pos = start_angle
+    gonio_axes[0].increment = increment
+    gonio_axes[0].num_steps = tot_imgs
+
+    scan = calculate_scan_points(gonio_axes[0],
+                                 rotation=True,
+                                 tot_num_imgs=tot_imgs)
+    goniometer = Goniometer(gonio_axes, scan)
+    osc, trans = goniometer.define_scan_from_goniometer_axes()
```

Comment:
Sorry, forgot a comment.

Must: remove this as it is not needed. When a scan is passed, the Goniometer is automatically updated. This particular function instead is only needed when you don't already have a scan and want the Goniometer to calculate it for you from the start_pos/increment/num_steps values of the axes.